### DISCUSSION TITLE
VM Exception while processing transaction when Purchase Plan

### DISCUSSION THREAD

**Discussion by suntzu93** [subquery/subql#915] [Frontier TestNet - Bug Reporting]
I'm trying to Purchase Plan but got error : 

`VM Exception while processing transaction: revert Indexing service is not available to establish agreements`

<img width="1915" alt="Screen Shot 2022-04-26 at 18 25 31" src="https://user-images.githubusercontent.com/5988002/165289850-fe21d50f-9055-46cf-afa6-ae47a0b3d2c1.png">

**Comment by suntzu93** [comment]
This issue because I forget to send Announce Ready ! Working now.

**Comment by jim-counter** [comment]
I was going to report the same. I still think a nicer error message would improve the experience. Something like 'Selected indexer is not ready to serve this plan yet.'
![image](https://user-images.githubusercontent.com/11335994/165601613-5daeaf62-5bfb-4903-9575-6801e8b1a8c2.png)

**Comment by lilian79511** [comment]
Thanks for the report. Will work on better description for the error msg

**Comment by lilian79511** [comment]
This issues is fixed by handling the error on the Plan column
![image](https://user-images.githubusercontent.com/46732651/166396717-f63c0364-b6f0-45a6-b011-ef9241415601.png)
